query/src/users/user_mgr.rs

```diff
@@ -36,6 +36,11 @@ impl UserApiProvider {
                         UserPrivilegeSet::available_privileges_on_global(),
                     );
                     user_info.option.set_all_flag();
+                } else if user.is_root() {
+                    return Err(ErrorCode::UnknownUser(format!(
+                        "'{}'@'{}'",
```

Comment:
1.  better remove the  `"default" | "" | "root"`  code
2.  a little more error message like `only accept root from localhost ` ?